### PR TITLE
[expanse-272] Expanse Expander - Behavior/certificate API support

### PR DESCRIPTION
**Pull request by andrew-expanse**
## Status
- [] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
N/A
## Description
Added Support for pulling behavior data to create new incidents, as well as behavior and certificate data for enrichment.

This PR has been recreated based on https://github.com/demisto/content/pull/6176 using a clean fork.

## Screenshots
N/A

## Minimum version of Demisto
- [ ] 4.5.0
- [x] 5.0.0
- [ ] 5.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

## Must have
- [x] Tests
- [x] Documentation 

## Demisto Partner?
- [x] The title must be in the following format: **[YOUR_PARTNER_ID] short description**

### CODE REVIEW

File: Packs/Expanse/Integrations/Expanse/Expanse.py [hunk 1]
```diff
@@ -123,6 +139,7 @@ def http_request(method, endpoint, params=None, token=False):
         return_error("No authorization token provided")
     head = make_headers(endpoint, token)
     url = make_url(endpoint)
+    demisto.info("Making request to {} with params: {}".format(url, params))
```
Comment (anara123) [inline Packs/Expanse/Integrations/Expanse/Expanse.py:142]:
```suggestion
    demisto.debug("Making request to {} with params: {}".format(url, params))
```

File: Packs/Expanse/Integrations/Expanse/Expanse.yml [hunk 2]
```diff
@@ -387,6 +391,201 @@ script:
...
+      isArray: true
+      name: common_name
+      required: true
+      secret: false
+    deprecated: false
+    description: certificate command
+    execution: false
+    name: certificate
+    outputs:
+    - contextPath: Certificate.SearchTerm
```
Comment (anara123) [inline Packs/Expanse/Integrations/Expanse/Expanse.yml:406]: delete the `Certificate.*` outputs/context paths. Leave just the `Expanse.Certificate.*`
We don't have Certificate standard reputation. All our standard outputs https://xsoar.pan.dev/docs/integrations/context-standards

File: Packs/Expanse/Integrations/Expanse/Expanse.yml [hunk 3]
```diff
@@ -387,6 +391,201 @@ script:
...
+    - default: true
+      description: domain to search
+      isArray: true
+      name: common_name
+      required: true
+      secret: false
+    deprecated: false
+    description: certificate command
+    execution: false
+    name: certificate
```
Comment (anara123) [inline Packs/Expanse/Integrations/Expanse/Expanse.yml:404]: change to `expanse-get-certificate`

File: Packs/Expanse/Integrations/Expanse/Expanse.yml [hunk 4]
```diff
@@ -387,6 +391,201 @@ script:
...
+    - default: true
+      description: ip to search
+      isArray: true
+      name: ip
+      required: true
+      secret: false
+    deprecated: false
+    description: Returns Risky Flows for specified IP (truncated to 20 results)
+    execution: false
+    name: behavior
```
Comment (anara123) [inline Packs/Expanse/Integrations/Expanse/Expanse.yml:560]: change the name to be `expanse-get-behavior`

File: Packs/Expanse/Integrations/Expanse/CHANGELOG.md [hunk 5]
```diff
@@ -2,6 +2,9 @@
   - Shortened the period of time that tokens are considered valid, to avoid authorization errors.
   - Fixed an issue related to the ***ip*** command where an error is generated if the API returns a partial response.
   - Added friendly values for various empty fields returned by the ***domain*** command.
+  - Added Support for pulling behavior data to create new incidents.
+  - Added Support for the **behavior** command.
```
Comment (anara123) [inline Packs/Expanse/Integrations/Expanse/CHANGELOG.md:6]:
```suggestion
  - Added Support for the **expanse-get-behavior** command.
```

File: Packs/Expanse/Integrations/Expanse/CHANGELOG.md [hunk 6]
```diff
@@ -2,6 +2,9 @@
...
+  - Added Support for the **certificate** command.
```
Comment (anara123) [inline Packs/Expanse/Integrations/Expanse/CHANGELOG.md:7]:
```suggestion
  - Added Support for the **expanse-get-certificate** command.
```

File: Packs/Expanse/Integrations/Expanse/Expanse.py [hunk 7]
```diff
@@ -571,6 +762,82 @@ def domain_command():
...
+    }
+    token = do_auth()
+    results = http_request('GET', 'behavior/risky-flows', params, token=token)
+    try:
+        behaviors = results['data']
+    except Exception:
+        demisto.results("No data found")
+        return
+
+    dbot_context = {
```
Comment (anara123) [inline Packs/Expanse/Integrations/Expanse/Expanse.py:824]: no need for dbot_Score, we don't have reputation for behavior nor certificate. List of standard outputs https://xsoar.pan.dev/docs/integrations/context-standards

File: Packs/Expanse/Integrations/Expanse/Expanse.py [hunk 8]
```diff
@@ -607,6 +874,12 @@ def main():
         elif active_command == 'domain':
             domain_command()
 
+        elif active_command == 'certificate':
```
Comment (anara123) [inline Packs/Expanse/Integrations/Expanse/Expanse.py:877]:
```suggestion
        elif active_command == 'expanse-get-certificate':
```

File: Packs/Expanse/Integrations/Expanse/Expanse.py [hunk 9]
```diff
@@ -607,6 +874,12 @@ def main():
...
+            certificate_command()
+
+        elif active_command == 'behavior':
```
Comment (anara123) [inline Packs/Expanse/Integrations/Expanse/Expanse.py:880]:
```suggestion
        elif active_command == 'expanse-get-behavior':
```

File: Packs/Expanse/Integrations/Expanse/Expanse.py [hunk 10]
```diff
@@ -571,6 +762,82 @@ def domain_command():
...
+
+
+def behavior_command():
+    """
+    searches by ip for behavior details from Expanse
+    """
+    search = demisto.args()['ip']
+
+    now = datetime.today()
+    time_range = datetime.strftime(now - timedelta(days=FIRST_RUN), "%Y-%m-%d")
```
Comment (anara123) [inline Packs/Expanse/Integrations/Expanse/Expanse.py:810]: you better add new argument for start time.
call it `created_after`

**Comment by anara123** [comment]
Hello @andrew-expanse 
Please fix the notes and let's have a demo session. 
@fvigo FYI

File: Packs/Expanse/Integrations/Expanse/Expanse.py [hunk 11]
```diff
@@ -571,6 +762,78 @@ def domain_command():
...
+        "Indicator": search,
+        "Type": "certificate",
+        "Vendor": "Expanse",
+        "Score": 0
+    }
+    cert_context = get_certificate_context(cert)
+    expanse_cert_context = get_expanse_certificate_context(cert)
+
+    ec = {
+        'DBotScore': dbot_context,
```
Comment (anara123) [inline Packs/Expanse/Integrations/Expanse/Expanse.py:794]: delete DBotScore. There is on DBotSCore calculated here

File: Packs/Expanse/Integrations/Expanse/Expanse.py [hunk 12]
```diff
@@ -571,6 +762,78 @@ def domain_command():
...
+        'Certificate(val.SearchTerm == obj.SearchTerm)': cert_context,
```
Comment (anara123) [inline Packs/Expanse/Integrations/Expanse/Expanse.py:795]: Delete this line too

File: Packs/Expanse/Integrations/Expanse/Expanse.py [hunk 13]
```diff
@@ -571,6 +762,78 @@ def domain_command():
...
+    try:
+        certs = results['data']
+    except Exception:
+        demisto.results("No data found")
+        return
+
+    cert = certs[0]  # just return the first one
+    cert['search'] = search
+
+    dbot_context = {
```
Comment (anara123) [inline Packs/Expanse/Integrations/Expanse/Expanse.py:784]: delete this var, no need for it

**Comment by anara123** [comment]
@andrew-expanse also please update the `Expanse/Integrations/Expanse/README.md` file. You added 2 new commands, both of these commands should be documented with examples and results.
You can follow https://xsoar.pan.dev/docs/integrations/integration-docs
https://xsoar.pan.dev/docs/integrations/doc-structure

Example:
https://github.com/demisto/content/tree/master/Packs/XForceExchange/Integrations/XFE_v2

**Comment by anara123** [comment]
image

**Comment by anara123** [comment]
image

**Comment by anara123** [comment]
https://github.com/demisto/content/blob/master/Packs/HelloWorld/Integrations/HelloWorld/HelloWorld.py#L370

**Comment by andrew-expanse** [comment]
@anara123 Do you know why the build is still failing? It appears to be an issue with the README file, though it's complaining about lines there were not modified.

**Comment by anara123** [comment]
@andrew-expanse it fails on README.md
`> 383 | <td parentName="tr" {...{"align":null}}>{`Tags: `}<br>{`AdditionalNotes: `}<br>{`PointsOfContact:`}</td>`
it was expected that `<br>` closing but got `</td>`

**Comment by anara123** [comment]
Just fix it manually in the README

**Comment by anara123** [comment]
@andrew-expanse just change <br> to <br/> 
I think it should solve the problem

**Comment by andrew-expanse** [comment]
> Just fix it manually in the README

I guess my confusion is that the line numbers reported in the error don't correspond to the contents of the README file. 

Line 383 is [here](https://github.com/demisto/content/pull/6237/files#diff-1947103c353735eb7c9ebd8266a92e91R383 - but the content of the error is referring to [this](https://github.com/demisto/content/pull/6237/files#diff-1947103c353735eb7c9ebd8266a92e91R254) human readable output. 

I tried a few different approaches here but the error persists. The line in question doesn't look any different than the surrounding lines, so I'm still not sure how to resolve. I'm also not totally sure what kind of validation is being run here.

**Comment by anara123** [comment]
@andrew-expanse see my updated comment above. Change `<br>` to `</br>`

**Comment by andrew-expanse** [comment]
Hi @anara123 sorry to keep bothering you. I've tried using `</br>` several different ways. 

Replacing the `<br>` with `</br>` in the instance above (end of line 383) did seem to fix that instance, but then the build started complaining about the same issue 2 lines down.  (https://circleci.com/gh/demisto/content/47153)

Once I replaced all ending instances of `<br>` with `</br>` the build failed again, this time with the same failure. (https://circleci.com/gh/demisto/content/47173) as above.

Finally I tried replacing all instances of `<br>` with `</br>`, which led to a build failure on the original line. (https://circleci.com/gh/demisto/content/47179)

I'm also seeing new test failures for seemingly unrelated tests:
- Tests/scripts/hook_validations/tests/test_utils_test.py::TestGetRemoteFile::test_get_remote_file_sanity
- Tests/scripts/hook_validations/tests/test_utils_test.py::TestGetRemoteFile::test_get_remote_file_origin

Do you have any guidance?

**Comment by anara123** [comment]
@andrew-expanse I fixed the README issue. changed all `<br>` to `<br />`
The PR is approved. Are you done here from your side?

**Comment by anara123** [comment]
@kirbles19 please review

**Comment by andrew-expanse** [comment]
@anara123 all set on my end. I really appreciate your help here.

**Comment by kirbles19** [comment]
@anara123 Done.

**Pull request merged**